modules/installation-mirror-repository.adoc

```diff
@@ -129,9 +129,9 @@ that you mirrored content for.
 .. Build a signature URI:
 +
 ----
-$ URI="https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release/${DIGEST/:/=}/<signature_name>" <1>
+$ URI="https://mirror.openshift.com/pub/openshift-v4/signatures/openshift/release/${DIGEST/:/=}/signature-<n>" <1>
 ----
-<1> Specify the name of the signature to store.
+<1> Specify the name of the signature to store. The signature name must be in the format of `signature-<n>`, where `<n>` is a number. If you have multiple signatures, the signature that includes the lowest number in its name is applied first, and other signatures are applied in increasing numerical order if one fails.
```

Comment:
nit: maybe:
> <1> Specify the name of the signature to store. The signature name must be in the format of `signature-<n>`, where `<n>` is a number beginning with 1. If a signature is invalid, increment `<n>` until you find a valid signature or the requested `signature-<n>` does not exist. 
Or some such.  That adds "start at 1" and drop "is applied first", because this is more about "has anyone I trust signed this image?" and not about "who did something first?".

Reply:
> ... If a signature is invalid, increment `<n>` until you find a valid signature or the requested `signature-<n>` does not exist.

Where do these other certificates come from? Is this a hidden step to say "If you get to `$ gpg --no-default-keyring --keyring ./temp.keyring --verify <signature_name>` and that doesn't work, then name the next one `signature_n+1`, and it'll just run through all the `signature_n` files in that directory until it finds a functioning one?

Reply:
> Where do these other certificates come from?

We don't have any at the moment, but may grow some during a future key-rotation or some such.  In that case, we'd have `signature-1` by the new key and `signature-2` by the old key (order insignificant).

> Is this a hidden step...

I was hoping that these docs made it explicit, but yeah "keep going until you find a satisfactory sig or run out of options" is the intended behavior.  Hopefully we can teach `oc` to do this soon...

Reply:
> > Where do these other certificates come from?
> 
> We don't have any at the moment, but may grow some during a future key-rotation or some such. In that case, we'd have `signature-1` by the new key and `signature-2` by the old key (order insignificant).

"We" meaning OCP devs or "we" meaning the person making these certs?

How and why would they have more than one, and why would they be checking multiples at this point? If I'm reading the process right, they generate a very specific certificate and then check it. Is it actually relevant that you can check multiples until you find a valid one?

> 
> > Is this a hidden step...
> 
> I was hoping that these docs made it explicit, but yeah "keep going until you find a satisfactory sig or run out of options" is the intended behavior. Hopefully we can teach `oc` to do this soon...
That's what I'm trying to figure out. I'm not seeing a point in the current flow where a user would have a failed certificate or multiple certificates. There's not an explicit step to make another one. I'm trying to figure out if I need to another step around line 173 and what it would say or if this is a cool detail that's not helpful in this flow.

Reply:
> ... or "we" meaning the person making these certs?

This one.

> Is it actually relevant that you can check multiples until you find a valid one?

Not at the moment, but I'd be nervous if our manual-verification docs only supported `signature-1` when all the software implementations were set up to walk available sigs.  But maybe that would be fine and we'll have `oc` tooling for this by the time it matters?

> I'm trying to figure out if I need to another step around line 173...

Yeah, if you think the wording here isn't enough, you'd want a "if the sig doesn't check out, increment `signature-<index>` , return to defining `URI`, and try again.

Reply:
> Yeah, if you think the wording here isn't enough, you'd want a "if the sig doesn't check out, increment `signature-<index>` , return to defining `URI`, and try again.

Yup. I'm adding a commit with another step. (If I took your suggestion as-is, I would expect to see a bug about the note not being clear enough and having to add the step later.) Thank you for working through this with me, and please let me know if you see more changes after the update.